sdk/database/dbplugin/v5/testing/test_helpers.go

```diff
@@ -22,19 +22,47 @@ func getRequestTimeout(t *testing.T) time.Duration {
 	return dur
 }
 
-func AssertInitialize(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
+// AssertInitializeCircleCiTest help to diagnose CircleCI failures within AssertInitialize for mssql tests failing
+// with "Failed to initialize: error verifying connection ..."
+func AssertInitializeCircleCiTest(t *testing.T, db dbplugin.Database, req dbplugin.InitializeRequest) dbplugin.InitializeResponse {
 	t.Helper()
+	maxAttempts := 5
+	var resp dbplugin.InitializeResponse
+	var err error
+
+	for i := 1; i <= maxAttempts; i++ {
+		resp, err = verifyInitialize(t, db, req)
+		switch {
+		case err != nil:
+			t.Logf("Failed AssertInitialize attempt: %d with error:\n%+v\n", i, err)
+			time.Sleep(1 * time.Second)
+		case i == 1:
+			return resp
+		default:
+			t.Fatalf("AssertInitialize worked the %d time around with a 1 second sleep, but failed originally", i)
+		}
+	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), getRequestTimeout(t))
-	defer cancel()
+	t.Fatalf("Failed to initialize: %+v", err)
```

Comment:
this will stop the test, but the next line has it returning the response. i believe that return line would not get hit

Reply:
You are absolutely correct, the return resp was just to get the code too compile. a return nil doesn't work either as the return type is a struct not a pointer to the struct. 

Perhaps a panic statement instead, not sure how to rework the code to make the intention clearer though.